Document CCR auto-follow patterns and searchable snapshots indices (#70863)
elastic/elasticsearch: docs/reference/ccr/auto-follow.asciidoc

a new index on the remote cluster matches the auto-follow pattern, a
corresponding follower index is added to the local cluster.

NOTE: auto-follow patterns creates follower indices only for opened, fully
assigned indices that match the specified pattern in the remote cluster. The

>  patterns should not match special indices like <<frozen-indices,frozen indices>>,

Reviewer: > should not

I wonder if this is too strict a wording in light of #67686, at least for searchable snapshots, which we will simply just ignore after #70580? Not sure of frozen and read-only though, I suppose we would fail on them (in which case the wording is fine for those).

Author: > Not sure of frozen and read-only though, I suppose we would fail on them (in which case the wording is fine for those).

Yes, auto-follow pattern picks up such indices but the follower indices quickly become red as they cannot be initialized (frozen) or cannot read remote index.

Author: I noticed that my first try would be more exact. James helped me to find the right wording and I pushed 4811d5c50adcb968ce1de91260803de4b7083bef. I removed the mention of frozen indices as they might be removed in the future and they are included in indices that blocks writes.

> NOTE: auto-follow patterns creates follower indices only for opened, fully
> assigned indices that match the specified pattern in the remote cluster. The
>  patterns should not match special indices like <<frozen-indices,frozen indices>>,
> <<searchable-snapshots,{search-snaps}>> or <<index-blocks-read-only, read-only indices>>
> that cannot be used for CCR purpose.

Reviewer 2: I'd streamline the wording a little:
```suggestion
NOTE: Auto-follow patterns only match regular open indices on the remote
cluster. Auto-follow patterns do not match indices that can't be used for
{ccr-init}, such as <<frozen-indices,frozen indices>>,
<<searchable-snapshots,{search-snaps}>>, or <<index-blocks-read-only,read-only
indices>>.
```

In the original copy, I wasn't sure what `fully assigned` meant. I _think_ this means the index's data is completely stored on local shards (as opposed to searchable snapshots using a shared cache with a repo). If that's the case, I think we can use `regular` index. If not, let me know.

Author: "fully assigned" is misleading, sorry, I should have written "with all primary shards started".

Reviewer 2: Thanks for the answer. I amended my suggestion.
